python/ray/train/tests/test_huggingface_trainer.py

```diff
@@ -167,6 +182,47 @@ def test_validation(ray_start_4_cpus):
         trainer.fit().error
 
 
+# Tests if checkpointing and restoring during tuning works correctly.
+def test_tune(ray_start_8_cpus):
```

Comment:
is there a smaller-scale unit test we can add for this behavior in addition to this full integration test?

Reply:
Also does this test produce the race condition every single time?

Reply:
I ran it three times and it failed reliably. As for a unit test, that may be tricky. I'll try to think of something.